Source/WebKit/UIProcess/API/APITargetedElementInfo.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "APITargetedElementInfo.h"
+
+#include "PageClient.h"
+#include "WebPageProxy.h"
+
+namespace API {
+using namespace WebKit;
+
+TargetedElementInfo::TargetedElementInfo(WebPageProxy& page, WebCore::TargetedElementInfo&& info)
+    : m_info(WTFMove(info))
+    , m_page(page)
+{
+}
+
+bool TargetedElementInfo::isSameElement(const TargetedElementInfo& other) const
+{
+    return m_info.elementIdentifier == other.m_info.elementIdentifier
+        && m_info.documentIdentifier == other.m_info.documentIdentifier;
+}
+
+WebCore::FloatRect TargetedElementInfo::boundsInWebView() const
+{
+    return m_page ? m_page->pageClient().rootViewToWebView(boundsInRootView()) : WebCore::FloatRect { };
```

Comment:
You should get a RefPtr<WebKit::WebPageProxy> on the stack, according to our smart pointer guidelines.
```suggestion
    RefPtr page = m_page.get();
    if (!page)
        return { };
    return page->pageClient().rootViewToWebView(boundsInRootView());
```